### ISSUE TITLE
StableLM doesn't work if `--quantize` isn't provided

### ISSUE THREAD

**Issue by vvsotnikov** [huggingface/text-generation-inference#347]
### System Info

After #277 was merged, StableLM models (and probably other GPT-NeoX models too, I haven't tested) stopped working if `--quantize` isn't provided.

### Information

- [ ] Docker
- [ ] The CLI directly

### Tasks

- [ ] An officially supported command
- [ ] My own modifications

### Reproduction

Run `docker run --rm --gpus all -p 8080:80 --pull always ghcr.io/huggingface/text-generation-inference:latest --model-id stabilityai/stablelm-tuned-alpha-3b`

### Expected behavior

Expected: inference API is working.
Actual behavior:
```
latest: Pulling from huggingface/text-generation-inference
Digest: sha256:4156c6be0ee27b4fe11678d52019b879cc2812151ccce027a620eb51bd98a4e2
Status: Image is up to date for ghcr.io/huggingface/text-generation-inference:latest
2023-05-21T06:07:46.819593Z  INFO text_generation_launcher: Args { model_id: "bigscience/bloom-560m", revision: None, sharded: None, num_shard: None, quantize: None, max_concurrent_requests: 128, max_best_of: 2, max_stop_sequences: 4, max_input_length: 1000, max_total_tokens: 1512, max_batch_size: None, waiting_served_ratio: 1.2, max_batch_total_tokens: 32000, max_waiting_tokens: 20, port: 80, shard_uds_path: "/tmp/text-generation-server", master_addr: "localhost", master_port: 29500, huggingface_hub_cache: Some("/data"), weights_cache_override: None, disable_custom_kernels: false, json_output: false, otlp_endpoint: None, cors_allow_origin: [], watermark_gamma: None, watermark_delta: None, env: false }
2023-05-21T06:07:46.819694Z  INFO text_generation_launcher: Starting download process.
2023-05-21T06:07:49.440985Z  INFO download: text_generation_launcher: Download file: model.safetensors

2023-05-21T06:08:06.939562Z  INFO download: text_generation_launcher: Downloaded /data/models--bigscience--bloom-560m/snapshots/4f42c91d806a19ae1a46af6c3fb5f4990d884cd6/model.safetensors in 0:00:17.

2023-05-21T06:08:06.939629Z  INFO download: text_generation_launcher: Download: [1/1] -- ETA: 0

2023-05-21T06:08:07.238247Z  INFO text_generation_launcher: Successfully downloaded weights.
2023-05-21T06:08:07.238421Z  INFO text_generation_launcher: Starting shard 0
2023-05-21T06:08:16.690119Z  INFO shard-manager: text_generation_launcher: Server started at unix:///tmp/text-generation-server-0
 rank=0
2023-05-21T06:08:16.747581Z  INFO text_generation_launcher: Shard 0 ready in 9.508774753s
2023-05-21T06:08:16.846916Z  INFO text_generation_launcher: Starting Webserver
2023-05-21T06:08:19.865876Z  INFO text_generation_router: router/src/main.rs:178: Connected
^C2023-05-21T06:08:20.951296Z  INFO text_generation_launcher: Waiting for webserver to gracefully shutdown
2023-05-21T06:08:20.951389Z  INFO text_generation_router::server: router/src/server.rs:704: signal received, starting graceful shutdown
2023-05-21T06:08:21.014792Z  INFO text_generation_launcher: Webserver terminated
2023-05-21T06:08:21.014819Z  INFO text_generation_launcher: Shutting down shards
2023-05-21T06:08:21.279366Z  INFO text_generation_launcher: Shard 0 terminated
➜  ~ docker run --rm --gpus all -p 8080:80 --pull always ghcr.io/huggingface/text-generation-inference:latest --model-id vvsotnikov/stablelm-tuned-alpha-3b-16bit
latest: Pulling from huggingface/text-generation-inference
Digest: sha256:4156c6be0ee27b4fe11678d52019b879cc2812151ccce027a620eb51bd98a4e2
Status: Image is up to date for ghcr.io/huggingface/text-generation-inference:latest
2023-05-21T06:16:08.954582Z  INFO text_generation_launcher: Args { model_id: "vvsotnikov/stablelm-tuned-alpha-3b-16bit", revision: None, sharded: None, num_shard: None, quantize: None, max_concurrent_requests: 128, max_best_of: 2, max_stop_sequences: 4, max_input_length: 1000, max_total_tokens: 1512, max_batch_size: None, waiting_served_ratio: 1.2, max_batch_total_tokens: 32000, max_waiting_tokens: 20, port: 80, shard_uds_path: "/tmp/text-generation-server", master_addr: "localhost", master_port: 29500, huggingface_hub_cache: Some("/data"), weights_cache_override: None, disable_custom_kernels: false, json_output: false, otlp_endpoint: None, cors_allow_origin: [], watermark_gamma: None, watermark_delta: None, env: false }
2023-05-21T06:16:08.954666Z  INFO text_generation_launcher: Starting download process.
2023-05-21T06:16:10.961998Z  INFO download: text_generation_launcher: Download file: model.safetensors

2023-05-21T06:18:03.853572Z  INFO download: text_generation_launcher: Downloaded /data/models--vvsotnikov--stablelm-tuned-alpha-3b-16bit/snapshots/7d177ff97909d25ce48065bbbb7f1cd8e4957587/model.safetensors in 0:01:52.

2023-05-21T06:18:03.853634Z  INFO download: text_generation_launcher: Download: [1/1] -- ETA: 0

2023-05-21T06:18:04.164226Z  INFO text_generation_launcher: Successfully downloaded weights.
2023-05-21T06:18:04.164396Z  INFO text_generation_launcher: Starting shard 0
2023-05-21T06:18:14.174025Z  INFO text_generation_launcher: Waiting for shard 0 to be ready...
2023-05-21T06:18:24.183511Z  INFO text_generation_launcher: Waiting for shard 0 to be ready...
2023-05-21T06:18:34.193080Z  INFO text_generation_launcher: Waiting for shard 0 to be ready...
2023-05-21T06:18:44.203103Z  INFO text_generation_launcher: Waiting for shard 0 to be ready...
2023-05-21T06:18:54.211721Z  INFO text_generation_launcher: Waiting for shard 0 to be ready...
2023-05-21T06:19:02.929745Z ERROR shard-manager: text_generation_launcher: Error when initializing model
Traceback (most recent call last):
  File "/opt/conda/bin/text-generation-server", line 8, in <module>
    sys.exit(app())
  File "/opt/conda/lib/python3.9/site-packages/typer/main.py", line 311, in __call__
    return get_command(self)(*args, **kwargs)
  File "/opt/conda/lib/python3.9/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/opt/conda/lib/python3.9/site-packages/typer/core.py", line 778, in main
    return _main(
  File "/opt/conda/lib/python3.9/site-packages/typer/core.py", line 216, in _main
    rv = self.invoke(ctx)
  File "/opt/conda/lib/python3.9/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/opt/conda/lib/python3.9/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/opt/conda/lib/python3.9/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/opt/conda/lib/python3.9/site-packages/typer/main.py", line 683, in wrapper
    return callback(**use_params)  # type: ignore
  File "/opt/conda/lib/python3.9/site-packages/text_generation_server/cli.py", line 66, in serve
    server.serve(model_id, revision, sharded, quantize, uds_path)
  File "/opt/conda/lib/python3.9/site-packages/text_generation_server/server.py", line 155, in serve
    asyncio.run(serve_inner(model_id, revision, sharded, quantize))
  File "/opt/conda/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/opt/conda/lib/python3.9/asyncio/base_events.py", line 634, in run_until_complete
    self.run_forever()
  File "/opt/conda/lib/python3.9/asyncio/base_events.py", line 601, in run_forever
    self._run_once()
  File "/opt/conda/lib/python3.9/asyncio/base_events.py", line 1905, in _run_once
    handle._run()
  File "/opt/conda/lib/python3.9/asyncio/events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
> File "/opt/conda/lib/python3.9/site-packages/text_generation_server/server.py", line 124, in serve_inner
    model = get_model(model_id, revision, sharded, quantize)
  File "/opt/conda/lib/python3.9/site-packages/text_generation_server/models/__init__.py", line 139, in get_model
    return neox_cls(model_id, revision, quantize=quantize)
  File "/opt/conda/lib/python3.9/site-packages/text_generation_server/models/flash_neox.py", line 32, in __init__
    super(FlashNeoX, self).__init__(
  File "/opt/conda/lib/python3.9/site-packages/text_generation_server/models/flash_causal_lm.py", line 407, in __init__
    model = model_cls.from_pretrained(
  File "/opt/conda/lib/python3.9/site-packages/text_generation_server/models/custom_modeling/flash_neox_modeling.py", line 469, in from_pretrained
    model.post_load_weights(load_in_8bit)
  File "/opt/conda/lib/python3.9/site-packages/text_generation_server/models/custom_modeling/flash_neox_modeling.py", line 458, in post_load_weights
    self.gpt_neox.post_load_weights(quantize)
  File "/opt/conda/lib/python3.9/site-packages/text_generation_server/models/custom_modeling/flash_neox_modeling.py", line 351, in post_load_weights
    layer.attention.query_key_value.prepare_weights(quantize)
  File "/opt/conda/lib/python3.9/site-packages/text_generation_server/utils/layers.py", line 57, in prepare_weights
    raise ValueError(f"Unexpected quantize `{quantize}`")
ValueError: Unexpected quantize `False`
 rank=0
Error: ShardCannotStart
2023-05-21T06:19:03.414552Z ERROR text_generation_launcher: Shard 0 failed to start:
Some weights of the model checkpoint at vvsotnikov/stablelm-tuned-alpha-3b-16bit were not used when initializing FlashGPTNeoXForCausalLM: ['gpt_neox.layers.0.attention.bias', 'gpt_neox.layers.9.attention.bias', 'gpt_neox.layers.1.attention.masked_bias', 'gpt_neox.layers.8.attention.masked_bias', 'gpt_neox.layers.13.attention.bias', 'gpt_neox.layers.15.attention.masked_bias', 'gpt_neox.layers.4.attention.bias', 'gpt_neox.layers.12.attention.masked_bias', 'gpt_neox.layers.11.attention.masked_bias', 'gpt_neox.layers.10.attention.masked_bias', 'gpt_neox.layers.3.attention.bias', 'gpt_neox.layers.6.attention.bias', 'gpt_neox.layers.7.attention.bias', 'gpt_neox.layers.2.attention.masked_bias', 'gpt_neox.layers.14.attention.masked_bias', 'gpt_neox.layers.13.attention.masked_bias', 'gpt_neox.layers.9.attention.masked_bias', 'gpt_neox.layers.0.attention.masked_bias', 'gpt_neox.layers.11.attention.bias', 'gpt_neox.layers.2.attention.bias', 'gpt_neox.layers.5.attention.bias', 'gpt_neox.layers.8.attention.bias', 'gpt_neox.layers.12.attention.bias', 'gpt_neox.layers.7.attention.masked_bias', 'gpt_neox.layers.5.attention.masked_bias', 'gpt_neox.layers.1.attention.bias', 'gpt_neox.layers.3.attention.masked_bias', 'gpt_neox.layers.15.attention.bias', 'gpt_neox.layers.4.attention.masked_bias', 'gpt_neox.layers.10.attention.bias', 'gpt_neox.layers.6.attention.masked_bias', 'gpt_neox.layers.14.attention.bias']
- This IS expected if you are initializing FlashGPTNeoXForCausalLM from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
- This IS NOT expected if you are initializing FlashGPTNeoXForCausalLM from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).
Traceback (most recent call last):

  File "/opt/conda/bin/text-generation-server", line 8, in <module>
    sys.exit(app())

  File "/opt/conda/lib/python3.9/site-packages/text_generation_server/cli.py", line 66, in serve
    server.serve(model_id, revision, sharded, quantize, uds_path)

  File "/opt/conda/lib/python3.9/site-packages/text_generation_server/server.py", line 155, in serve
    asyncio.run(serve_inner(model_id, revision, sharded, quantize))

  File "/opt/conda/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)

  File "/opt/conda/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
    return future.result()

  File "/opt/conda/lib/python3.9/site-packages/text_generation_server/server.py", line 124, in serve_inner
    model = get_model(model_id, revision, sharded, quantize)

  File "/opt/conda/lib/python3.9/site-packages/text_generation_server/models/__init__.py", line 139, in get_model
    return neox_cls(model_id, revision, quantize=quantize)

  File "/opt/conda/lib/python3.9/site-packages/text_generation_server/models/flash_neox.py", line 32, in __init__
    super(FlashNeoX, self).__init__(

  File "/opt/conda/lib/python3.9/site-packages/text_generation_server/models/flash_causal_lm.py", line 407, in __init__
    model = model_cls.from_pretrained(

  File "/opt/conda/lib/python3.9/site-packages/text_generation_server/models/custom_modeling/flash_neox_modeling.py", line 469, in from_pretrained
    model.post_load_weights(load_in_8bit)

  File "/opt/conda/lib/python3.9/site-packages/text_generation_server/models/custom_modeling/flash_neox_modeling.py", line 458, in post_load_weights
    self.gpt_neox.post_load_weights(quantize)

  File "/opt/conda/lib/python3.9/site-packages/text_generation_server/models/custom_modeling/flash_neox_modeling.py", line 351, in post_load_weights
    layer.attention.query_key_value.prepare_weights(quantize)

  File "/opt/conda/lib/python3.9/site-packages/text_generation_server/utils/layers.py", line 57, in prepare_weights
    raise ValueError(f"Unexpected quantize `{quantize}`")

ValueError: Unexpected quantize `False`


2023-05-21T06:19:03.414597Z  INFO text_generation_launcher: Shutting down shards
➜  ~ docker run --rm --gpus all -p 8080:80 --pull always ghcr.io/huggingface/text-generation-inference:latest --model-id stabilityai/stablelm-tuned-alpha-3b
latest: Pulling from huggingface/text-generation-inference
Digest: sha256:4156c6be0ee27b4fe11678d52019b879cc2812151ccce027a620eb51bd98a4e2
Status: Image is up to date for ghcr.io/huggingface/text-generation-inference:latest
2023-05-21T06:21:12.097166Z  INFO text_generation_launcher: Args { model_id: "stabilityai/stablelm-tuned-alpha-3b", revision: None, sharded: None, num_shard: None, quantize: None, max_concurrent_requests: 128, max_best_of: 2, max_stop_sequences: 4, max_input_length: 1000, max_total_tokens: 1512, max_batch_size: None, waiting_served_ratio: 1.2, max_batch_total_tokens: 32000, max_waiting_tokens: 20, port: 80, shard_uds_path: "/tmp/text-generation-server", master_addr: "localhost", master_port: 29500, huggingface_hub_cache: Some("/data"), weights_cache_override: None, disable_custom_kernels: false, json_output: false, otlp_endpoint: None, cors_allow_origin: [], watermark_gamma: None, watermark_delta: None, env: false }
2023-05-21T06:21:12.097232Z  INFO text_generation_launcher: Starting download process.
2023-05-21T06:21:15.193934Z  WARN download: text_generation_launcher: No safetensors weights found for model stabilityai/stablelm-tuned-alpha-3b at revision None. Downloading PyTorch weights.

2023-05-21T06:21:15.367639Z  INFO download: text_generation_launcher: Download file: pytorch_model-00001-of-00002.bin

2023-05-21T06:24:02.560947Z  INFO download: text_generation_launcher: Downloaded /data/models--stabilityai--stablelm-tuned-alpha-3b/snapshots/d1c03d2114451d562416b9efe4281d319ceff99e/pytorch_model-00001-of-00002.bin in 0:02:47.

2023-05-21T06:24:02.561043Z  INFO download: text_generation_launcher: Download: [1/2] -- ETA: 0:02:47

2023-05-21T06:24:02.561324Z  INFO download: text_generation_launcher: Download file: pytorch_model-00002-of-00002.bin

2023-05-21T06:25:50.282095Z  INFO download: text_generation_launcher: Downloaded /data/models--stabilityai--stablelm-tuned-alpha-3b/snapshots/d1c03d2114451d562416b9efe4281d319ceff99e/pytorch_model-00002-of-00002.bin in 0:01:47.

2023-05-21T06:25:50.282163Z  INFO download: text_generation_launcher: Download: [2/2] -- ETA: 0

2023-05-21T06:25:50.282263Z  WARN download: text_generation_launcher: No safetensors weights found for model stabilityai/stablelm-tuned-alpha-3b at revision None. Converting PyTorch weights to safetensors.

2023-05-21T06:25:50.282413Z  INFO download: text_generation_launcher: Convert /data/models--stabilityai--stablelm-tuned-alpha-3b/snapshots/d1c03d2114451d562416b9efe4281d319ceff99e/pytorch_model-00001-of-00002.bin to /data/models--stabilityai--stablelm-tuned-alpha-3b/snapshots/d1c03d2114451d562416b9efe4281d319ceff99e/model-00001-of-00002.safetensors.

2023-05-21T06:28:05.487768Z  INFO download: text_generation_launcher: Convert: [1/2] -- Took: 0:02:15.205169

2023-05-21T06:28:05.487841Z  INFO download: text_generation_launcher: Convert /data/models--stabilityai--stablelm-tuned-alpha-3b/snapshots/d1c03d2114451d562416b9efe4281d319ceff99e/pytorch_model-00002-of-00002.bin to /data/models--stabilityai--stablelm-tuned-alpha-3b/snapshots/d1c03d2114451d562416b9efe4281d319ceff99e/model-00002-of-00002.safetensors.


2023-05-21T06:28:36.770847Z  INFO download: text_generation_launcher: Convert: [2/2] -- Took: 0:00:31.282476

2023-05-21T06:28:38.218138Z  INFO text_generation_launcher: Successfully downloaded weights.
2023-05-21T06:28:38.394035Z  INFO text_generation_launcher: Starting shard 0
2023-05-21T06:28:49.107083Z  INFO text_generation_launcher: Waiting for shard 0 to be ready...
2023-05-21T06:28:59.119121Z  INFO text_generation_launcher: Waiting for shard 0 to be ready...
2023-05-21T06:29:09.129403Z  INFO text_generation_launcher: Waiting for shard 0 to be ready...
2023-05-21T06:29:19.139081Z  INFO text_generation_launcher: Waiting for shard 0 to be ready...
2023-05-21T06:29:29.149474Z  INFO text_generation_launcher: Waiting for shard 0 to be ready...
2023-05-21T06:29:39.159781Z  INFO text_generation_launcher: Waiting for shard 0 to be ready...
2023-05-21T06:29:49.169980Z  INFO text_generation_launcher: Waiting for shard 0 to be ready...
2023-05-21T06:29:59.179994Z  INFO text_generation_launcher: Waiting for shard 0 to be ready...
2023-05-21T06:30:09.190246Z  INFO text_generation_launcher: Waiting for shard 0 to be ready...
2023-05-21T06:30:19.200743Z  INFO text_generation_launcher: Waiting for shard 0 to be ready...
2023-05-21T06:30:29.210183Z  INFO text_generation_launcher: Waiting for shard 0 to be ready...
2023-05-21T06:30:39.220430Z  INFO text_generation_launcher: Waiting for shard 0 to be ready...
2023-05-21T06:30:49.231779Z  INFO text_generation_launcher: Waiting for shard 0 to be ready...
2023-05-21T06:30:52.120547Z ERROR shard-manager: text_generation_launcher: Error when initializing model
Traceback (most recent call last):
  File "/opt/conda/bin/text-generation-server", line 8, in <module>
    sys.exit(app())
  File "/opt/conda/lib/python3.9/site-packages/typer/main.py", line 311, in __call__
    return get_command(self)(*args, **kwargs)
  File "/opt/conda/lib/python3.9/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/opt/conda/lib/python3.9/site-packages/typer/core.py", line 778, in main
    return _main(
  File "/opt/conda/lib/python3.9/site-packages/typer/core.py", line 216, in _main
    rv = self.invoke(ctx)
  File "/opt/conda/lib/python3.9/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/opt/conda/lib/python3.9/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/opt/conda/lib/python3.9/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/opt/conda/lib/python3.9/site-packages/typer/main.py", line 683, in wrapper
    return callback(**use_params)  # type: ignore
  File "/opt/conda/lib/python3.9/site-packages/text_generation_server/cli.py", line 66, in serve
    server.serve(model_id, revision, sharded, quantize, uds_path)
  File "/opt/conda/lib/python3.9/site-packages/text_generation_server/server.py", line 155, in serve
    asyncio.run(serve_inner(model_id, revision, sharded, quantize))
  File "/opt/conda/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/opt/conda/lib/python3.9/asyncio/base_events.py", line 634, in run_until_complete
    self.run_forever()
  File "/opt/conda/lib/python3.9/asyncio/base_events.py", line 601, in run_forever
    self._run_once()
  File "/opt/conda/lib/python3.9/asyncio/base_events.py", line 1905, in _run_once
    handle._run()
  File "/opt/conda/lib/python3.9/asyncio/events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
> File "/opt/conda/lib/python3.9/site-packages/text_generation_server/server.py", line 124, in serve_inner
    model = get_model(model_id, revision, sharded, quantize)
  File "/opt/conda/lib/python3.9/site-packages/text_generation_server/models/__init__.py", line 139, in get_model
    return neox_cls(model_id, revision, quantize=quantize)
  File "/opt/conda/lib/python3.9/site-packages/text_generation_server/models/flash_neox.py", line 32, in __init__
    super(FlashNeoX, self).__init__(
  File "/opt/conda/lib/python3.9/site-packages/text_generation_server/models/flash_causal_lm.py", line 407, in __init__
    model = model_cls.from_pretrained(
  File "/opt/conda/lib/python3.9/site-packages/text_generation_server/models/custom_modeling/flash_neox_modeling.py", line 469, in from_pretrained
    model.post_load_weights(load_in_8bit)
  File "/opt/conda/lib/python3.9/site-packages/text_generation_server/models/custom_modeling/flash_neox_modeling.py", line 458, in post_load_weights
    self.gpt_neox.post_load_weights(quantize)
  File "/opt/conda/lib/python3.9/site-packages/text_generation_server/models/custom_modeling/flash_neox_modeling.py", line 351, in post_load_weights
    layer.attention.query_key_value.prepare_weights(quantize)
  File "/opt/conda/lib/python3.9/site-packages/text_generation_server/utils/layers.py", line 57, in prepare_weights
    raise ValueError(f"Unexpected quantize `{quantize}`")
ValueError: Unexpected quantize `False`
 rank=0
2023-05-21T06:30:53.840219Z ERROR text_generation_launcher: Shard 0 failed to start:
Loading checkpoint shards: 100%|██████████| 2/2 [00:44<00:00, 22.12s/it]
Some weights of the model checkpoint at stabilityai/stablelm-tuned-alpha-3b were not used when initializing FlashGPTNeoXForCausalLM: ['gpt_neox.layers.13.attention.masked_bias', 'gpt_neox.layers.1.attention.bias', 'gpt_neox.layers.11.attention.bias', 'gpt_neox.layers.15.attention.masked_bias', 'gpt_neox.layers.5.attention.bias', 'gpt_neox.layers.11.attention.masked_bias', 'gpt_neox.layers.8.attention.bias', 'gpt_neox.layers.13.attention.bias', 'gpt_neox.layers.9.attention.masked_bias', 'gpt_neox.layers.6.attention.bias', 'gpt_neox.layers.3.attention.masked_bias', 'gpt_neox.layers.10.attention.bias', 'gpt_neox.layers.1.attention.masked_bias', 'gpt_neox.layers.2.attention.bias', 'gpt_neox.layers.12.attention.bias', 'gpt_neox.layers.14.attention.bias', 'gpt_neox.layers.10.attention.masked_bias', 'gpt_neox.layers.15.attention.bias', 'gpt_neox.layers.5.attention.masked_bias', 'gpt_neox.layers.12.attention.masked_bias', 'gpt_neox.layers.3.attention.bias', 'gpt_neox.layers.14.attention.masked_bias', 'gpt_neox.layers.4.attention.masked_bias', 'gpt_neox.layers.0.attention.bias', 'gpt_neox.layers.7.attention.bias', 'gpt_neox.layers.7.attention.masked_bias', 'gpt_neox.layers.6.attention.masked_bias', 'gpt_neox.layers.2.attention.masked_bias', 'gpt_neox.layers.8.attention.masked_bias', 'gpt_neox.layers.0.attention.masked_bias', 'gpt_neox.layers.4.attention.bias', 'gpt_neox.layers.9.attention.bias']
- This IS expected if you are initializing FlashGPTNeoXForCausalLM from the checkpoint of a model trained on another task or with another architecture (e.g. initializing a BertForSequenceClassification model from a BertForPreTraining model).
- This IS NOT expected if you are initializing FlashGPTNeoXForCausalLM from the checkpoint of a model that you expect to be exactly identical (initializing a BertForSequenceClassification model from a BertForSequenceClassification model).
Traceback (most recent call last):

  File "/opt/conda/bin/text-generation-server", line 8, in <module>
    sys.exit(app())

  File "/opt/conda/lib/python3.9/site-packages/text_generation_server/cli.py", line 66, in serve
    server.serve(model_id, revision, sharded, quantize, uds_path)

  File "/opt/conda/lib/python3.9/site-packages/text_generation_server/server.py", line 155, in serve
    asyncio.run(serve_inner(model_id, revision, sharded, quantize))

  File "/opt/conda/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)

  File "/opt/conda/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
    return future.result()

  File "/opt/conda/lib/python3.9/site-packages/text_generation_server/server.py", line 124, in serve_inner
    model = get_model(model_id, revision, sharded, quantize)

  File "/opt/conda/lib/python3.9/site-packages/text_generation_server/models/__init__.py", line 139, in get_model
    return neox_cls(model_id, revision, quantize=quantize)

  File "/opt/conda/lib/python3.9/site-packages/text_generation_server/models/flash_neox.py", line 32, in __init__
    super(FlashNeoX, self).__init__(

  File "/opt/conda/lib/python3.9/site-packages/text_generation_server/models/flash_causal_lm.py", line 407, in __init__
    model = model_cls.from_pretrained(

  File "/opt/conda/lib/python3.9/site-packages/text_generation_server/models/custom_modeling/flash_neox_modeling.py", line 469, in from_pretrained
    model.post_load_weights(load_in_8bit)

  File "/opt/conda/lib/python3.9/site-packages/text_generation_server/models/custom_modeling/flash_neox_modeling.py", line 458, in post_load_weights
    self.gpt_neox.post_load_weights(quantize)

  File "/opt/conda/lib/python3.9/site-packages/text_generation_server/models/custom_modeling/flash_neox_modeling.py", line 351, in post_load_weights
    layer.attention.query_key_value.prepare_weights(quantize)

  File "/opt/conda/lib/python3.9/site-packages/text_generation_server/utils/layers.py", line 57, in prepare_weights
    raise ValueError(f"Unexpected quantize `{quantize}`")

ValueError: Unexpected quantize `False`


2023-05-21T06:30:53.840598Z  INFO text_generation_launcher: Shutting down shards
Error: ShardCannotStart

**Comment by hyunwoongko** [comment]
Same error.

**Comment by gsaivinay** [comment]
I got the same error... I think it is because of the recent changes done to allow GPTQ as quantization method. To workaround this issue until it is officially fixed, I modified code as below.

[here](https://github.com/huggingface/text-generation-inference/blob/main/server/text_generation_server/utils/layers.py#L28)
`if quantize == "bitsandbytes":` to `if quantize == "bitsandbytes" or quantize == True:`

[here](https://github.com/huggingface/text-generation-inference/blob/main/server/text_generation_server/utils/layers.py#L54) 
`elif quantize is None:` to `elif quantize is None or quantize == False:`
